src/components/special/_toggle.less

```diff
@@ -39,24 +41,67 @@ input[type="checkbox"].toggle-btn {
     top: 0;
     z-index: -1;
   }
+
   &:checked:before {
     transform: @toggleSwitchOnPosition;
   }
+
   &:checked:after {
     background-color: @toggleOnBgColor;
   }
+
+  &.xsmall-btn {
+    width: @toggleWidthSmall;
+    height: @toggleHeightSmall;
+    border-radius: @toggleBorderRadiusSmall;
+
+    &.xsmall-btn:checked:before {
+      transform: @toggleSwitchOnPositionSmall;
+    }
+
+    &.xsmall-btn:before {
+      height: calc(@toggleWidthSmall / 2.3);
+      width: calc(@toggleWidthSmall / 2.25);
+      transform: @toggleSwitchOffPositionSmall;
+    }
+
+    &.toggle-on-btn {
+      &:before {
+        transform: @toggleSwitchOnPositionSmall;
+      }
+
+      &:before {
+        transform: @toggleSwitchOffPositionSmall;
```

Comment:
this is causing problems as it is applying the active styling for before even when it's active so please remove it.